### PR TITLE
KNOV-2434 abrak

### PR DESCRIPTION
**Pull request by dashkun1215**
check

### CODE REVIEW

File: abrakadabra.yaml [hunk 1]
```diff
@@ -154,11 +154,101 @@ Resources:
...
+          - Effect: Allow
+            Action:
+            - cloudformation:ValidateTemplate
+            Resource: "*"
+          - Effect: Allow
+            Action:
+            - appconfig:StartDeployment
+            - appconfig:StopDeployment
+            - appconfig:GetDeployment
+            Resource: "*"
```
Comment (okostiuk667) [inline abrakadabra.yaml:251]: Need to remove *, here should be added direct resources...

File: abrakadabra.yaml [hunk 2]
```diff
@@ -154,11 +154,101 @@ Resources:
...
+            Effect: Allow
           - Effect: Allow
-            Action: "*"
+            Action:
+            - servicecatalog:ListProvisioningArtifacts
+            - servicecatalog:CreateProvisioningArtifact
+            - servicecatalog:DescribeProvisioningArtifact
+            - servicecatalog:DeleteProvisioningArtifact
+            - servicecatalog:UpdateProduct
             Resource: "*"
```
Comment (okostiuk667) [inline abrakadabra.yaml:241]: Need to remove *, here should be added direct resources...

File: abrakadabra.yaml [hunk 3]
```diff
@@ -154,11 +154,101 @@ Resources:
...
+          - Action:
+            - opsworks:CreateDeployment
+            - opsworks:DescribeApps
+            - opsworks:DescribeCommands
+            - opsworks:DescribeDeployments
+            - opsworks:DescribeInstances
+            - opsworks:DescribeStacks
+            - opsworks:UpdateApp
+            - opsworks:UpdateStack
+            Resource: "*"
```
Comment (okostiuk667) [inline abrakadabra.yaml:219]: Need to remove *, here should be added direct resources...

File: abrakadabra.yaml [hunk 4]
```diff
@@ -154,11 +154,101 @@ Resources:
...
+            - codecommit:GetUploadArchiveStatus
+            Resource: "*"
+            Effect: Allow
+          - Action:
+            - elasticbeanstalk:*
+            - ec2:*
+            - cloudwatch:*
+            - s3:*
+            - cloudformation:*
+            Resource: "*"
```
Comment (okostiuk667) [inline abrakadabra.yaml:208]: Need to remove *, here should be added direct resources...

File: abrakadabra.yaml [hunk 5]
```diff
@@ -154,11 +154,101 @@ Resources:
...
+                - ec2.amazonaws.com
+                - ecs-tasks.amazonaws.com
+          - Action:
+            - codecommit:GetBranch
+            - codecommit:GetCommit
+            - codecommit:GetRepository
+            - codecommit:GetUploadArchiveStatus
+            Resource: "*"
+            Effect: Allow
+          - Action:
```
Comment (okostiuk667) [inline abrakadabra.yaml:202]: Need to remove *, here should be added direct actions...

File: abrakadabra.yaml [hunk 6]
```diff
@@ -154,11 +154,101 @@ Resources:
...
+                - cloudformation.amazonaws.com
+                - elasticbeanstalk.amazonaws.com
+                - ec2.amazonaws.com
+                - ecs-tasks.amazonaws.com
+          - Action:
+            - codecommit:GetBranch
+            - codecommit:GetCommit
+            - codecommit:GetRepository
+            - codecommit:GetUploadArchiveStatus
+            Resource: "*"
```
Comment (okostiuk667) [inline abrakadabra.yaml:200]: Need to remove *, here should be added direct resources...

File: abrakadabra.yaml [hunk 7]
```diff
@@ -154,11 +154,101 @@ Resources:
...
+            Resource:
+              - !GetAtt KnovioInfrastructureCodeBuildLogGroup.Arn
+          - Action:
+              - logs:DescribeLogGroups
+            Effect: Allow
+            Resource:
+              - !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:*"
+          - Action:
+            - iam:PassRole
+            Resource: "*"
```
Comment (okostiuk667) [inline abrakadabra.yaml:186]: Need to remove *, here should be added direct resources...

File: abrakadabra.yaml [hunk 8]
```diff
@@ -154,11 +154,101 @@ Resources:
...
+            Resource:
+              - !Join [ '/', [ !GetAtt KnovioInfrastructureCodePipelineBucket.Arn, '*' ] ]
+              - Fn::Sub:
+                  - "${DevOpsBucketArn}/*"
+                  - DevOpsBucketArn:
+                      Fn::ImportValue:
+                        !Sub ${AccountType}:knovio:s3:devops:bucket:arn
+              - Fn::ImportValue: !Sub ${AccountType}:knovio:s3:devops:bucket:arn
+          - Action:
+              - logs:*
```
Comment (okostiuk667) [inline abrakadabra.yaml:175]: Need to remove *, here should be added directly actions...

File: abrakadabra.yaml [hunk 9]
```diff
@@ -154,11 +154,101 @@ Resources:
       PolicyDocument:
         Version: '2012-10-17'
         Statement:
+          - Sid: "CodePipelineEnvBucketsAccess"
+            Action:
+              - s3:*
```
Comment (okostiuk667) [inline abrakadabra.yaml:159]: Need to remove *, here should be added directly actions...

**Review by okostiuk667 — COMMENTED** [review]
Also, I see that you used one role and one policy everywhere, you should split all policies and roles, one resource = one role.

**Comment by okostiuk667** [comment]
also, you have conflicts in file abrakadabra.yaml please fix it....)

File: abrakadabra.yaml (hunk 1 above)
Comment (dashkun1215) [reply abrakadabra.yaml]: ok

File: abrakadabra.yaml (hunk 2 above)
Comment (dashkun1215) [reply abrakadabra.yaml]: ok

File: abrakadabra.yaml (hunk 7 above)
Comment (dashkun1215) [reply abrakadabra.yaml]: ok

File: abrakadabra.yaml (hunk 8 above)
Comment (dashkun1215) [reply abrakadabra.yaml]: ok

File: abrakadabra.yaml (hunk 9 above)
Comment (dashkun1215) [reply abrakadabra.yaml]: ok

**Pull request merged**